src/main/java/net/cardosi/mojo/tools/J2cl.java

```diff
@@ -1,56 +1,72 @@
-                        .map(File::getAbsolutePath)
-                        .collect(Collectors.toList())
-                )
-                .setOutput(jsOutDir.toPath())
-                .setEmitReadableLibraryInfo(false)
-                .setEmitReadableSourceMap(false)
-                .setGenerateKytheIndexingMetadata(false);
+        args.add("-d");
+        args.add(jsOutDir.toPath().toString());
+        args.add("-cp");
+        args.add(Stream.concat(Stream.of(bootstrap), strippedClasspath.stream())
+                       .map(File::getAbsolutePath)
+                       .collect(Collectors.joining("" + File.pathSeparatorChar)));
     }
 
-    public boolean transpile(List<FrontendUtils.FileInfo> sourcesToCompile, List<FrontendUtils.FileInfo> nativeSources) {
-        J2clTranspilerOptions options = optionsBuilder
-                .setSources(sourcesToCompile)
-                .setNativeSources(nativeSources)
-                .build();
+    public boolean transpile(List<Path> sourcesToCompile, List<Path> nativeSources) {
+        ImmutableList.Builder<String> commandLineArgsBuilder =
+                ImmutableList.<String>builder();
+
+        commandLineArgsBuilder.addAll(sourcesToCompile.stream().map(Path::toString).collect(Collectors.toList()));
+
+        if (!nativeSources.isEmpty()) {
+            commandLineArgsBuilder.add("-nativesourcepath",
+                                       nativeSources.stream().map(Path::toString)
+                                                    .collect(Collectors.joining("" + File.pathSeparatorChar)));
+        }
+
+        commandLineArgsBuilder.addAll(args);
+
         Problems problems;
+        ImmutableList<String> cmdLine = commandLineArgsBuilder.build();
+        //System.out.println("j2cl: " + cmdLine);
         try {
-            problems = J2clTranspiler.transpile(options);
-        } catch (Throwable t) {
-            System.out.println(options);
-            throw t;
+            problems = transpile(cmdLine);
+        } catch (Exception e) {
+            System.out.println(cmdLine);
+            throw new RuntimeException(e);
         }
+
 //        if (problems.hasErrors() || problems.hasWarnings()) {
 //            problems.getErrors().forEach(System.out::println);
 //            problems.getWarnings().forEach(System.out::println);
 //        } else {
-//            problems.getInfoMessages().forEach(System.out::println);
+//           // problems.getInfoMessages().forEach(System.out::println);
 //        }
-        problems.getMessages().forEach(System.out::println);
         if (problems.hasErrors()) {
-            System.out.println(options);
+            System.out.println(cmdLine);
         }
         return !problems.hasErrors();
     }
+
+    private static Problems transpile(Iterable<String> args) throws Exception {
+        // J2clCommandLineRunner.run is hidden since we don't want it to be used as an entry point. As a
```

Comment:
:thinking: `J2clCommandLineRunner.run` is public, and what you should be using here. Above, the `problems` are directly printed to `System.out` (why not `System.err` btw?) and otherwise `hasErrors()` si called to determine if the processing was successful. With `J2clCommandLineRunner.run` you get exactly the same behavior (except `System.err` instead of `System.out`, and it prints a summary of the number of errors and warnings) but get an error code instead, i.e. `1`/`0` instead of `false`/`true` (see https://github.com/google/j2cl/blob/1128b5a882e918975d300abb1fd96aa24be430a8/transpiler/java/com/google/j2cl/common/CommandLineTool.java#L64-L67; `runForTest` directly invokes `processRequest`)

I.e. you can replace lines 45–70 with a simple
```java
int status = J2clCommandLineRunner.run(cmdLine);
if (status != 0) {
  System.out.println(cmdLine);
}
return (status == 0);
```

Reply:
done.